lint.tsconfig.json

```diff
@@ -1,4 +1,16 @@
 {
-  "extends": "./tsconfig.json",
```

Comment:
Might it cause problems down the road that linting and the project's default tsconfig aren't tied together at all? I.e. tests are succeeding because of one tsconfig configuration, whereas the actual library fails because it doesn't have the same constraints?

Reply:
Great point, pretty much all of this is done in a way that's trying to get all the repo's linting in the same place.  I tried to leave the build steps mostly unchanged.  I removed this here to accomplish that, but when we move to the monorepo we could try to setup a shared tsconfig